src/downzip-sw.js

```diff
@@ -83,25 +83,34 @@ self.addEventListener('fetch', async (event) => {
             // Append all the downloaded data
             try {
                 await new Promise((resolve, reject) => {
-                    fetch(file.downloadUrl).then(response => response.body).then(async (stream) => {
-                        const reader = stream.getReader()
-                        let doneReading = false
-                        while (!doneReading) {
-                            const chunk = await reader.read()
-                            const {done, value} = chunk
-
-                            if (done) {
-                                // If this stream has finished, resolve and return
+                    try {

```

Comment:
Isn't a Promise already a try..catch block? (implicit) This feels redundant.

I think you could just do:
```js
try { // big try block here
  // No other try catch block here
  await new Promise(resolve => {
    fetch().then(/* do stuff */)
  })
  // No `.catch()` block here unless you would like to handle a specific exception class
} catch (e) {
  // catch everything here
}
```

Any unhandled error in either fetch call or `then` block, will bubble up outside of the promise, into the global `catch` block.

Reply:
@pandaiolo actually, this won't catch rejected promises inside `then`, I'll only handle it if the outer Promise rejects